### DISCUSSION TITLE
Strange components in HTML source

### DISCUSSION THREAD

**Discussion by morningstarsabrina** [zauberzeug/nicegui#377] [Q&A] [answered]
I was analyzing the html source of a simple  app and I noticed some strange components like joystick component

```
<html>
<body>
<!--StartFragment-->

var joystick = app.component('joystick', {template:'#tpl-joys',
--
      var joystick = app.component('joystick', {template:'#tpl-joys',
  mounted() {
    const joystick = nipplejs.create({
      zone: this.$el,
      position: { left: "50%", top: "50%" },
      dynamicPage: true,
      ...this.options,
    });
    joystick.on("start", (e) => this.$emit("start", e));
    joystick.on("move", (_, data) => this.$emit("move", { data }));
    joystick.on("end", (e) => this.$emit("end", e));
  },
  props: {
    options: Object,
  },
});

```

is this normal ?

**Comment by falkoschindler** [comment, marked as answer]
Yes, that's normal. Because NiceGUI doesn't know whether the Python code might lead to a `ui.joystick` element being created eventually during runtime, the necessary JavaScript code is already loaded at the beginning. You can avoid that be explicitly excluding it with `ui.run(..., exclude='joystick')`.

But we're already thinking about using a lazier approach for loading such dependencies (#241). The `ui.chart` component, for example, needs so many JavaScript files that we import them only when needed.

By the way: `ui.joystick` is somewhat special because it is implemented as a *.vue file instead of *.js. These are served embedded in the index HTML instead of importing them as a separate resource.

**Reply by morningstarsabrina** [reply]
ok  I get it. thanks
